packages/multichain-network-controller/src/MultichainNetworkService.ts

```diff
@@ -0,0 +1,68 @@
+import { is } from '@metamask/superstruct';
+import type { CaipAccountId } from '@metamask/utils';
+
+import {
+  MULTICHAIN_ACCOUNTS_CLIENT_HEADER,
+  MULTICHAIN_ACCOUNTS_CLIENT_ID,
+} from './constants';
+import {
+  ActiveNetworksResponseStruct,
+  type ActiveNetworksResponse,
+} from './types';
+import { buildActiveNetworksUrl, assertCaipAccountIds } from './utils';
+
+/**
+ * Service responsible for fetching network activity data from the API.
+ */
+export class MultichainNetworkService {
+  readonly #fetch: typeof fetch;
+
+  constructor({ fetch: fetchFunction }: { fetch: typeof fetch }) {
+    this.#fetch = fetchFunction;
+  }
+
+  /**
+   * Fetches active networks for the given account IDs.
+   *
+   * @param accountIds - Array of CAIP-10 account IDs to fetch activity for.
+   * @returns Promise resolving to the active networks response.
+   * @throws Error if the response format is invalid or the request fails.
+   */
+  async fetchNetworkActivity(
+    accountIds: CaipAccountId[],
+  ): Promise<ActiveNetworksResponse> {
+    try {
+      const url = buildActiveNetworksUrl(accountIds);
+
+      const response = await this.#fetch(url.toString(), {
+        method: 'GET',
+        headers: {
+          [MULTICHAIN_ACCOUNTS_CLIENT_HEADER]: MULTICHAIN_ACCOUNTS_CLIENT_ID,
+          Accept: 'application/json',
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+
+      const data: unknown = await response.json();
+
+      if (!is(data, ActiveNetworksResponseStruct)) {
+        throw new Error('Invalid response format from active networks API');
+      }
+      const typedData = data as ActiveNetworksResponse;
+      assertCaipAccountIds(typedData.activeNetworks);
```

Comment:
After re-reviewing, I believe we could get away without this helper 🤔 if we use `CaipAccountIdStruct` in `ActiveNetworksResponseStruct`, we would have the guarantee that `activeNetworks` will contains on valid identifiers.

But, if you need to have checks against specific namespace/addresses/references, then no, we would need some sort of other logic there.

Reply:
As discussed together, we can get rid of this function and rely only on `superstruct.assert`.